lib/plug/upload.ex

```diff
@@ -161,7 +161,7 @@ defmodule Plug.Upload do
   defp i(integer), do: Integer.to_string(integer)
 
   defp path(prefix, tmp) do
-    {_mega, sec, mili} = :erlang.now
+    {_mega, sec, mili} = :os.timestamp
```

Comment:
What about keeping `:os.timestamp` and adding `<> "-" <> i(:erlang.system_info(:scheduler_id))`? Would fulfil all goals and be fully compatible with 17 and 18?